attributes/ms_dotnet4.rb

```diff
@@ -57,7 +57,7 @@
     # Starting with windows 8 and Server 2012 old version of .NET Framework 4 are builtin or included as feature
     if nt_version >= 6.2
       require 'chef/win32/version'
-      if node['kernel']['os_info']['product_type'] != Chef::ReservedNames::Win32::Version::VER_NT_WORKSTATION
+      if node['kernel']['os_info']['product_type'] == Chef::ReservedNames::Win32::Version::VER_NT_WORKSTATION
```

Comment:
It is really not clear what is going on with version in this file. Can we have comments explaining a bit more what is each constant or number used?